BraintreeCore/src/main/java/com/braintreepayments/api/PaymentMethodNonce.java

```diff
@@ -16,26 +15,61 @@
     private static final String PAYMENT_METHOD_TYPE_KEY = "type";
     private static final String PAYMENT_METHOD_NONCE_KEY = "nonce";
     private static final String PAYMENT_METHOD_DEFAULT_KEY = "default";
+    private static final String DESCRIPTION_KEY = "description";
 
     private final String nonce;
     private final boolean isDefault;
+    private final String typeLabel;
+    private final String description;
 
-    private @PaymentMethodType final int mType;
+    private @PaymentMethodType final int type;
 
     static PaymentMethodNonce fromJSON(JSONObject inputJson) throws JSONException {
         String typeString = inputJson.getString(PAYMENT_METHOD_TYPE_KEY);
         int type = paymentMethodTypeFromString(typeString);
 
         String nonce = inputJson.getString(PAYMENT_METHOD_NONCE_KEY);
         boolean isDefault = inputJson.optBoolean(PAYMENT_METHOD_DEFAULT_KEY, false);
+        String description = inputJson.optString(DESCRIPTION_KEY);
```

Comment:
Just for the sake of clarity and consistency with iOS, I wonder if we should stop using the `description` field from the Gateway altogether and just set the description based on the actual property being used (i.e., email address for PayPal, username for Venmo, etc.).

It feels a little odd to me that right now we're using the Gateway's `description` for some payment method types but not others.

Reply:
Yes good call - we aligned with iOS and now return email for PayPal, username for Venmo, and last four for Card. Since Google Pay can be a card or PayPal account we decided to just return "Google Pay" for description, and since Visa Checkout seems to only have last two in the response we decided to return "Visa Checkout".

Reply:
Sounds good! So we don't need to pull the description from the Gateway response anymore, right?
```suggestion
        String description;
```